week2/length.c

```diff
@@ -0,0 +1,80 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include "length.h"
+
+//Function Definitions
+Input GetUserInput(char* input)
+{
+    //initialize struct Input 
+    Input userInput;
+    
+    userInput.stringLength = strlen(input);
+    
+    char* actualString = malloc(sizeof(*actualString) * userInput.stringLength);
```

Comment:
Seems like an off-by-one error here - strlen only returns the number of *characters* in the string, so if you want to create a new string, you need to allocate the proper number of bytes + one byte for null terminator. If you step through in the debugger, your strcpy call must be putting in the null terminator over the last character in your input string